src/gl/shim.c

```diff
@@ -113,16 +114,109 @@ int64_t glXSwapBuffersMscOML(void* dpy, void* drawable, int64_t target_msc, int6
     loadMangoHud();
 
     // Get the hooked glXSwapBuffersMscOML function from the loaded library if available
-    int64_t (*pglXSwapBuffersMscOML)(void*, void*, int64_t, int64_t, int64_t) = dlsym(handle, "glXSwapBuffersMscOML");
+    int64_t (*pglXSwapBuffersMscOML)(void*, void*, int64_t, int64_t, int64_t) = real_dlsym(handle, "glXSwapBuffersMscOML");
     if (pglXSwapBuffersMscOML) {
         return pglXSwapBuffersMscOML(dpy, drawable, target_msc, divisor, remainder);
     } else {
         // Fall back to the original glXSwapBuffersMscOML function
-        pglXSwapBuffersMscOML = dlsym(RTLD_NEXT, "glXSwapBuffersMscOML");
+        pglXSwapBuffersMscOML = real_dlsym(RTLD_NEXT, "glXSwapBuffersMscOML");
         if (pglXSwapBuffersMscOML) {
             return pglXSwapBuffersMscOML(dpy, drawable, target_msc, divisor, remainder);
         }
     }
 
     return -1;
 }
+
+void* glXGetProcAddress(const unsigned char* procName)
+{
+    loadMangoHud();
+
+    void* (*pglXGetProcAddress)(const unsigned char*) = real_dlsym(handle, "glXGetProcAddress");
+    if (pglXGetProcAddress) {
+        return pglXGetProcAddress(procName);
+    } else {
+        pglXGetProcAddress = real_dlsym(RTLD_NEXT, "glXGetProcAddress");
+        if (pglXGetProcAddress) {
+            return pglXGetProcAddress(procName);
+        }
+    }
+
+    return NULL;
+}
+
+void* glXGetProcAddressARB(const unsigned char* procName)
+{
+    loadMangoHud();
+
+    void* (*pglXGetProcAddressARB)(const unsigned char*) = real_dlsym(handle, "glXGetProcAddressARB");
+    if (pglXGetProcAddressARB) {
+        return pglXGetProcAddressARB(procName);
+    } else {
+        pglXGetProcAddressARB = real_dlsym(RTLD_NEXT, "glXGetProcAddressARB");
+        if (pglXGetProcAddressARB) {
+            return pglXGetProcAddressARB(procName);
+        }
+    }
+
+    return NULL;
+}
+
+void* eglGetProcAddress(const char *procName)
+{
+    loadMangoHud();
+
+    void* (*peglGetProcAddress)(const char*) = real_dlsym(handle, "eglGetProcAddress");
+    if (peglGetProcAddress) {
+        return peglGetProcAddress(procName);
+    } else {
+        peglGetProcAddress = real_dlsym(RTLD_NEXT, "eglGetProcAddress");
+        if (peglGetProcAddress) {
+            return peglGetProcAddress(procName);
+        }
+    }
+
+    return NULL;
+}
+
+struct func_ptr {
+    const char* name;
+    void* ptr;
+};
+
+#define ADD_HOOK(fn) { #fn, (void*)fn }
+static struct func_ptr hooks[] = {
+    ADD_HOOK(glXGetProcAddress),
+    ADD_HOOK(glXGetProcAddressARB),
+    ADD_HOOK(glXSwapBuffers),
+    ADD_HOOK(glXSwapBuffersMscOML),
+    ADD_HOOK(eglSwapBuffers),
+    ADD_HOOK(eglGetPlatformDisplay),
+    ADD_HOOK(eglGetDisplay),
+    ADD_HOOK(eglGetProcAddress)
+};
+#undef ADD_HOOK
+
+#ifndef ARRAY_SIZE
+#define ARRAY_SIZE(arr) sizeof(arr)/sizeof(arr[0])
+#endif
+
+void* dlsym(void *handle, const char *name)
+{
+    const char* dlsym_enabled = getenv("MANGOHUD_DLSYM");
```

Comment:
The thing is we can't know if the user set `--dlsym` in the mangohud script.
Maybe it would make more sense to just use dlsym by default and add an env for disabling it?

Reply:
that's what it does lol

Reply:
So then `--dlsym` does nothing. I guess we could just retire that?

Reply:
yeah it's been removed from the mangohud script (I didn't update the README in this PR, I probably should)